app/services/beta/search/perform_search_service.rb

```diff
@@ -10,7 +10,7 @@ def initialize(query, filters = {})
       end
 
       def call
-        retries = 0
+        retries ||= 0
```

Comment:
Important: without this the retries is reset every time (and does not keep track of the retries)